src/ucp/core/ucp_ep.c

```diff
@@ -1196,6 +1196,10 @@ static void ucp_ep_discard_lanes(ucp_ep_h ep, ucs_status_t discard_status)
     ucp_lane_index_t lane;
     uct_ep_h uct_ep;
 
+    if (ep->flags & UCP_EP_FLAG_FAILED) {
```

Comment:
/* Avoid calling ucp_ep_discard_lanes_callback() that will purge ep's requests, if 
  we already started discard and purge process for this ep. Doing so could complete 
  send requests before UCT lanes using them are flushed. */